apps/web/screens/Grants/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,470 @@
+                                      new Date(grant.dateTo),
+                                      'dd.MM.yyyy',
+                                    )}`,
+                                  }
+                                : null,
+                              {
+                                icon: 'time' as const,
+                                //todo: fix when the text is ready
+                                text: 'Frestur til 16.08.2024, kl. 15.00',
+                              },
+                              grant.categoryTag?.title
+                                ? {
+                                    icon: 'informationCircle' as const,
+                                    text: grant.categoryTag.title,
+                                  }
+                                : undefined,
+                            ].filter(isDefined)}
+                          />
+                        )}
+                      </Box>
+                    )
+                  })}
+                </Inline>
+              ) : undefined}
+              {!grants?.length && !error && !loading && (
+                <Box
+                  display="flex"
+                  alignItems="center"
+                  justifyContent="center"
+                  background="white"
+                  borderWidth="standard"
+                  borderRadius="xl"
+                  borderColor="blue200"
+                  flexDirection={['columnReverse', 'columnReverse', 'row']}
+                  columnGap={[2, 4, 8, 8, 20]}
+                  paddingY={[5, 8]}
+                  paddingX={[3, 3, 5, 10]}
+                  rowGap={[7, 7, 0]}
+                >
+                  <Box display="flex" flexDirection="column" rowGap={1}>
+                    <Text variant={'h3'} as={'h3'} color="dark400">
+                      {formatMessage(m.search.noResultsFound)}
+                    </Text>
+                    <Text whiteSpace="preLine">
+                      Borem ipsum dolor sit amet, consectetur adipiscing elit.
+                    </Text>
+                  </Box>
+                  <img
+                    width="240"
+                    src="/assets/sofa.svg"
+                    alt={formatMessage(m.search.noResultsFound)}
+                  />
+                </Box>
+              )}
+            </Box>
+          )}
+        </SidebarLayout>
+      </Box>
+    </GrantWrapper>
+  )
+}
+
+interface GrantsHomeProps {
+  locale: Locale
+  initialGrants?: Array<Grant>
+  tags?: Array<GenericTag>
+}
+
+const GrantsSearchResults: CustomScreen<GrantsHomeProps> = ({
+  initialGrants,
+  tags,
+  customPageData,
+  locale,
+}) => {
+  return (
+    <GrantsSearchResultsPage
+      initialGrants={initialGrants}
+      tags={tags}
+      locale={locale}
+      customPageData={customPageData}
+    />
+  )
+}
+
+GrantsSearchResults.getProps = async ({ apolloClient, locale, query }) => {
+  const queryPage = query?.page
```

Comment:
So you don't have to go through the whole fiasco of checking if it's an array you could also test out https://github.com/47ng/nuqs?tab=readme-ov-file#using-parsers-in-server-components which is a package added by Júní a few months ago that can help out in parsing query params and more.

Reply:
thats cool, i'll try it out